src/libxtp/tools/mol2orb.cc

```diff
@@ -0,0 +1,272 @@
+#include "mol2orb.h"
+
+// Third party includes
+#include <boost/algorithm/string.hpp>
+
+// Local VOTCA includes
+#include "votca/xtp/ecpaobasis.h"
+#include "votca/xtp/orbitals.h"
+#include <votca/tools/constants.h>
+
+// Local VOTCA includes
+#include "votca/xtp/logger.h"
+
+namespace votca {
+namespace xtp {
+
+void Mol2Orb::Initialize(const tools::Property& user_options) {
+  tools::Property options =
+      LoadDefaultsAndUpdateWithUserOptions("xtp", user_options);
+
+  _job_name = options.ifExistsReturnElseReturnDefault<std::string>("job_name",
+                                                                   _job_name);
+  _moldenfile = _job_name + ".molden.input";
+  _orbfile = _job_name + ".orb";
+  _xyzfile = _job_name + ".xyz";
+
+  _basisset_name = options.get(".basisset").as<std::string>();
+  _aux_basisset_name = options.get(".auxbasisset").as<std::string>();
+}
+
+void Mol2Orb::setupBasis(Orbitals& orbitals) {
+  // We only need the basis set for checking correctness
+  BasisSet bs;
+  bs.Load(_basisset_name);
+
+  // fill DFT AO basis by going through all atoms
+  _basis.Fill(bs, orbitals.QMAtoms());
+  orbitals.setDFTbasisName(_basisset_name);
+  XTP_LOG(Log::error, _log)
+      << TimeStamp() << " Basisset size: " << _basis.AOBasisSize()
+      << std::flush;
+  orbitals.setBasisSetSize(_basis.AOBasisSize());
+  orbitals.setAuxbasisName(_aux_basisset_name);
+}
+
+inline std::string Mol2Orb::readAtoms(QMMolecule& mol, std::string units,
+                                      std::ifstream& input_file) const {
+  std::string line;
+  std::istringstream iss(" ");
+  while (std::getline(input_file, line)) {
+    boost::trim(line);
+    if (line == "" || line[0] == '[') {
+      return line;
+    }
+    iss.str(line);
+    iss.clear();
+
+    // extract data
+    double x, y, z;
+    int atom_id;
+    std::string junk;
+    std::string atom_type;
+    iss >> atom_type >> atom_id >> junk >> x >> y >> z;
+    atom_id =
+        atom_id - 1;  // molden uses indexing from 1, we use indexing from 0
+
+    // Add data to orbitals object
+    using Atom =
+        typename std::iterator_traits<QMMolecule::iterator>::value_type;
+    Eigen::Vector3d pos(x, y, z);
+    if (units == "Angs") {
+      pos = tools::conv::ang2bohr * pos;
+    }
+    mol.push_back(Atom(atom_id, atom_type, pos));
+  }
+  return "";
```

Comment:
why not `void` then

Reply:
This is just in case, in principle it will never get here but the parser will find a new section to read with this piece of code 

``` 
if (line == "" || line[0] == '[') {
      return line;
    }
```

and return that section, in case that it is the last section in the molden file it returns an empty string.